truffle/src/main/java/org/jruby/truffle/nodes/core/ExceptionNodes.java

```diff
@@ -114,6 +115,7 @@ public BacktraceNode(RubyContext context, SourceSection sourceSection) {
             readCustomBacktrace = new ReadHeadObjectFieldNode("@custom_backtrace");
         }
 
+        @TruffleBoundary
```

Comment:
`asRubyStringArray()` should probably be the boundary, the rest should not have too many nodes.